### ISSUE TITLE
Avoid long time caching of maintenance pages

### ISSUE THREAD

**Issue by mkalkbrenner** [backdrop/backdrop-issues#85]
If the system is in maintenance mode, the maintenance page containing the site off-line message must not be cached. Neither in the internal page cache nor by external caches like varnish.

If you currently enable page caching for anonymous users, the system keeps on delivering cached site off-line message when the site is already back on-line again.

In the original d.o issue https://drupal.org/node/1032936 someone argued that it makes sense to cache this pages and to flush the page cache when leaving the maintenance mode. But this doesn't clear transparent caching proxies or browser caches.
And I assume that a site administrator won't put his site in maintenance mode during high traffic hours.
Therefor the simple solution is to not cache these pages at all.

**Comment by quicksketch** [comment]
@jenlampton made a new PR a few weeks ago at https://github.com/backdrop/backdrop/pull/594. It's looking good but isn't merging currently. I left a few items of feedback in there.

**Comment by quicksketch** [comment]
Latest PR is at https://github.com/backdrop/backdrop/pull/1188.

**Comment by quicksketch** [comment]
After looking at the recent pull requests, I realized the current approach also apparently doesn't actually set a shorter Cache-Control header, making it so that reverse proxies will be getting the full long cache lifetime.

In addition, the current approach is going to cause us some grief once #1214 (background-fetch) is implemented. If we allowed the maintenance page in the page cache, then the background fetch functionality might end up serving a maintenance page *up to a day* after maintenance mode had been turned off! Though the next immediate page load would be the full site again, it would be terribly confusing.

So I reworked the PR at https://github.com/backdrop/backdrop/pull/1188 to take a different approach:

- Maintenance pages are *never* stored in the database, per the original request. It also prevents them from having negative side-effects once background fetch is implemented.
- Maintenance pages have a default 10 second "Cache-Control" header, which will be used by browsers and any reverse proxies like Varnish.
- If all caching wants to be disabled entirely (including reverse proxies) during maintenance mode, the setting in `system.core.json` can be modified.

This makes it so that sites that are currently handled by Apache alone (and thus will be used to that amount of traffic hitting it) will serve maintenance page from scratch each time. Sites that have a reverse proxy will have a 10 second lifetime, with no cache-overlap caused by Backdrop's internal page cache.

This gets us the best situation possible. Heavy traffic sites are likely to have a reverse proxy and will be protected during downtime. Sites without a reverse proxy will need to generate the maintenance page each request, but it's light enough it shouldn't be an issue. All sites will have at maximum a 10 second delay coming out of maintenance mode.
